bin/npg_publish_illumina_run.pl

```diff
@@ -98,8 +98,9 @@
 }
 
 my $irods = WTSI::NPG::iRODS->new;
+my $wh_schema = WTSI::DNAP::Warehouse::Schema->connect();
```

Comment:
We have to be very careful here not to create a connection where it is not needed because the script can be running in the environment where it is impossible to get the db credentials or even connect to the db server.

I believe it would be fair to create a connection only if the driver type is set and contains the string `ml_warehouse` (all drivers of this type are defined in this package). It is possible to constraint the code further by getting all available driver types for mlwh and validating against this list.

Reply:
We also might find that when the driver type is not one of ml_warehouse drivers, setting the db schema leads to a failure in creating the driver object. Even if we do not observe the failure now, we can easily get it if the implementation of other driver types or of the `st::api::lims` object itself changes.